src/main/java/org/cascadebot/cascadebot/commands/music/SearchCommand.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.commands.music;
+
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Message;
+import net.dv8tion.jda.core.entities.TextChannel;
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.commandmeta.Argument;
+import org.cascadebot.cascadebot.commandmeta.ArgumentType;
+import org.cascadebot.cascadebot.commandmeta.CommandContext;
+import org.cascadebot.cascadebot.commandmeta.ICommandMain;
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.music.MusicHandler;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.utils.buttons.Button;
+import org.cascadebot.cascadebot.utils.buttons.ButtonGroup;
+import org.cascadebot.cascadebot.utils.buttons.IButtonRunnable;
+
+import java.util.Set;
+
+public class SearchCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if(context.getArgs().length < 1) {
+            context.getUIMessaging().replyUsage(this);
+            return;
+        }
+
+        CascadeBot.INS.getMusicHandler().searchTracks(context.getMessage(0), context.getChannel(), searchResults -> {
+            ButtonGroup buttonGroup = new ButtonGroup(sender.getUser().getIdLong(), context.getGuild().getIdLong());
+            int i = 0;
+            StringBuilder messageBuilder = new StringBuilder();
+            for(MusicHandler.SearchResult result : searchResults) {
+                i++;
+                char unicode = (char) (0x0030 + i);
+                buttonGroup.addButton(new Button.UnicodeButton(unicode + "\u20E3", (runner, channel, message) -> {
+                    if (!runner.equals(buttonGroup.getOwner())) {
+                        return;
+                    }
+                    message.delete().queue();
+                    context.getData().getMusicPlayer().loadLink(result.getUrl(), nothing -> {
```

Comment:
Should make a method handling this because this exact thing will be used in play

Reply:
I feel like it's better for the command to handle it so they can have better flexibility